flowzone.yml

```diff
@@ -1398,6 +1433,233 @@ jobs:
+      matrix:
+        target: ${{ fromJSON(needs.project_types.outputs.cargo_targets) }}
+
+    steps:
+      - *downloadSourceArtifact
+      - *extractSourceArtifact
+
+      - name: Set up toolchain ${{ matrix.target }}
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+          target: ${{ matrix.target }}
+
+      - name: Build release for toolchain ${{ matrix.target }}
+        uses: actions-rs/cargo@v1
+        with:
+          use-cross: true
+          command: test
+          args: --release --target ${{ matrix.target }} --manifest-path ${{ inputs.working_directory }}/Cargo.toml
+
+      - name: Install LLVM
+        run: sudo apt-get install -y llvm
+
+      - name: LLVM strip
+        run: llvm-strip target/${{ matrix.target }}/release/${{ needs.cargo_test.outputs.package }}
+
+      - name: Compress
+        run: |
+          tar -czvf ${{ needs.cargo_test.outputs.package }}-${{ matrix.target }}-v${{ needs.cargo_test.outputs.version }}.tar.gz -C target/${{ matrix.target }}/release ${{ needs.cargo_test.outputs.package }}
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: gh-release-${{ github.event.pull_request.head.sha || github.event.head_commit.id }}
+          path: ${{ needs.cargo_test.outputs.package }}-${{ matrix.target }}-v${{ needs.cargo_test.outputs.version }}.tar.gz
+          retention-days: 90
+
+  # INFO: cargo_finalize is not needed for now as github publishing is done by
+  # github_finalize. Once we add support for publishing to crates.io it will
+  # become necessary
+
+  ###################################################
+  # GitHub
+  ###################################################
+
+  github_publish:
+    name: Create draft Github release
+    runs-on: ${{ fromJSON(inputs.runs_on) }}
+    timeout-minutes: ${{ fromJSON(inputs.jobs_timeout_minutes) }}
+    needs:
+      - project_types
+      - versioned_source
+      - npm_publish
+      - cargo_publish
+      - custom_publish
+
+    if: |
+      !failure() && !cancelled() &&
+      needs.event_types.outputs.do_draft == 'true' &&
+      inputs.publish_to_github == true
+
+    defaults:
+      run:
+        working-directory: .
+        shell: bash --noprofile --norc -eo pipefail -x {0}
+
+    steps:
+      - name: Download release artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: gh-release-${{ github.event.pull_request.head.sha || github.event.head_commit.id }}
+          path: ${{ runner.temp }}
+
+      - name: Publish artifacts
+        uses: softprops/action-gh-release@v1
+        with:
+          name: ${{ github.event.pull_request.head.ref }}
+          tag_name: ${{ github.event.pull_request.head.ref }}
+          draft: true
+          prerelease: true
+          # Publish anything added to with the gh-release name
+          files: ${{ runner.temp }}/*
```

Comment:
During my testing, this action doesn't re-create a release if one already exists. So if the release has assets from previous PR runs, each subsequent run will add to these (not replace). Which is why I ended up having to delete previous draft releases manually.

Reply:
Oh, good catch, I'll add that

Reply:
@ab77 what about the artifacts? Do you know if artifacts uploaded to action context with the same name persist across PR runs?

Reply:
Nevermind, I just saw that it doesn't